lib/models/ecs-model-deployer.ts

```diff
@@ -0,0 +1,68 @@
+/**
+  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+  Licensed under the Apache License, Version 2.0 (the "License").
+  You may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+import { Construct } from 'constructs';
+import { Code, Function, Runtime } from 'aws-cdk-lib/aws-lambda';
+import { Role, ServicePrincipal, ManagedPolicy } from 'aws-cdk-lib/aws-iam';
+import { Stack, Duration, Size } from 'aws-cdk-lib';
+
+import { createCdkId } from '../core/utils';
+import { BaseProps, Config } from '../schema';
+
+export type ECSModelDeployerProps = {
+    vpcId: string;
+    securityGroupId: string;
+    config: Config;
+} & BaseProps;
+
+export class ECSModelDeployer extends Construct {
+    constructor (scope: Construct, id: string, props: ECSModelDeployerProps) {
+        super(scope, id);
+        const stackName = Stack.of(scope).stackName;
+        const role = new Role(this, createCdkId([stackName, 'ecs-model-deployer-role']), {
+            assumedBy: new ServicePrincipal('lambda.amazonaws.com')
+        });
+
+        role.addManagedPolicy(ManagedPolicy.fromAwsManagedPolicyName('AdministratorAccess'));
+
+        const stripped_config = {
+            'appName': props.config.appName,
+            'deploymentName': props.config.deploymentName,
+            'region': props.config.region,
+            'deploymentStage': props.config.deploymentStage,
+            'removalPolicy': props.config.removalPolicy,
+            's3BucketModels': props.config.s3BucketModels,
+            'mountS3DebUrl': props.config.mountS3DebUrl
+        };
+
+        const functionId = createCdkId([stackName, 'ecs_model_deployer']);
+        new Function(this, functionId, {
+            functionName: functionId,
+            runtime: Runtime.NODEJS_18_X,
+            handler: 'index.handler',
+            code: Code.fromAsset('./ecs_model_deployer/dist/'),
+            timeout: Duration.minutes(10),
```

Comment:
Since it seems like we're going to wait for the whole model deployment to happen, should we just give the maximum wait time of 15 minutes?

Reply:
We're actually not waiting for the entire stack creation/destruction. Right now I have it configured to wait 1 minute after starting the CDK command, then exit. In my testing, this has been enough time for the `create-stack` operation to hit cloudformation, so we can safely exit without affecting the actual deployment. I wanted to poll against `describe-stack` but unfortunately adding `@aws-sdk/client-cloudformation` pushed us over the 250MB limit.

Reply:
Ah I see, not a problem. That seems fine then. 10 minutes may be a lot, but if we're not at risk of the lambda cutting off before the operations have completed, then I'm good.